internal/pkg/aws/ecs/errors.go

```diff
@@ -3,7 +3,39 @@
 
 package ecs
 
-import "errors"
+import (
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+const (
+	fmtErrContainerStopped = "task %s: %s"
+)
 
 // ErrNoDefaultCluster occurs when the default cluster is not found.
 var ErrNoDefaultCluster = errors.New("default cluster does not exist")
+
+// ErrResourceNotReadyForTasks contains the STOPPED reason for the container of the first task that failed to start.
+type ErrWaiterResourceNotReadyForTasks struct {
+	tasks []*Task
+	awsErrResourceNotReady error
+}
+
+func (e *ErrWaiterResourceNotReadyForTasks) Error() string {
+	for _, task := range e.tasks {
+		if aws.StringValue(task.LastStatus) != DesiredStatusStopped {
+			continue
+		}
+
+		container := task.Containers[0] // NOTE: right now we only support one container per task
+		if aws.StringValue(container.LastStatus) == DesiredStatusStopped {
+			taskID, err := task.taskID(aws.StringValue(task.TaskArn))
+			if err != nil {
+				continue
+			}
+			return fmt.Sprintf(fmtErrContainerStopped, taskID[:shortTaskIDLength], aws.StringValue(container.Reason))
```

Comment:
We may also want to add a TODO to list the failure reason that any of the tasks stopped. (collect the reasons, join them together and return that)

Reply:
I was originally doing that. My concern was that the error messages could get very long as the number of tasks increase. Meanwhile the failure reasons for all tasks are likely the same in most cases because they all use the same configuration (task def, vpc, etc.).